calypso/csadmin/clicontracts/read.go

```diff
@@ -0,0 +1,149 @@
+package clicontracts
+
+import (
+	"bytes"
+	"encoding/hex"
+	"errors"
+	"fmt"
+	"io"
+	"os"
+
+	"go.dedis.ch/cothority/v3"
+	"go.dedis.ch/cothority/v3/byzcoin"
+	"go.dedis.ch/cothority/v3/byzcoin/bcadmin/lib"
+	"go.dedis.ch/cothority/v3/calypso"
+	"go.dedis.ch/cothority/v3/darc"
+	"go.dedis.ch/kyber/v3"
+	"go.dedis.ch/protobuf"
+	"gopkg.in/urfave/cli.v1"
+)
+
+// ReadSpawn spawns an instance of a read contract. This contract uses the darc
+// of the Write contract, which means that the signer of this transaction must
+// be allowed by the owner of the targeted write contract. The Write instance is
+// specified by the --instid argument. By default, the function uses the public
+// key of the signer to encrypt the requested data. However, a different public
+// key can be given a an hexadecimal string representation with --key.
+// With the --export option, the instance id is sent to STDOUT.
+func ReadSpawn(c *cli.Context) error {
+	bcArg := c.String("bc")
+	if bcArg == "" {
+		return errors.New("--bc flag is required")
+	}
+
+	cfg, cl, err := lib.LoadConfig(bcArg)
+	if err != nil {
+		return err
+	}
+
+	var signer *darc.Signer
+
+	sstr := c.String("sign")
+	if sstr == "" {
+		signer, err = lib.LoadKey(cfg.AdminIdentity)
+	} else {
+		signer, err = lib.LoadKeyFromString(sstr)
+	}
+	if err != nil {
+		return errors.New("failed to parse the signer: " + err.Error())
+	}
+
+	instidstr := c.String("instid")
+	if instidstr == "" {
+		return errors.New("please provide the write instance ID with --instid")
+	}
+
+	instidbuf, err := hex.DecodeString(instidstr)
+	if err != nil {
+		return errors.New("failed to decode instance id: " + err.Error())
+	}
+
+	pr, err := cl.GetProofFromLatest(instidbuf)
+	if err != nil {
+		return errors.New("couldn't get proof: " + err.Error())
+	}
+	proof := pr.Proof
+
+	exist, err := proof.InclusionProof.Exists(instidbuf)
+	if err != nil {
+		return errors.New("error while checking if proof exist: " + err.Error())
+	}
+	if !exist {
+		return errors.New("proof not found")
+	}
+
+	match := proof.InclusionProof.Match(instidbuf)
+	if !match {
+		return errors.New("proof does not match")
+	}
+
```

Comment:
since the proof has the value inside, you can also check whether the instance is a write instance